trinity/extensibility/plugin_manager.py

```diff
@@ -116,65 +190,77 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
+
+        self._logger.info("Shutting down PluginManager with scope %s", type(self._scope))
+
         for plugin in self._started_plugins:
+
+            if not isinstance(plugin, BaseSyncStopPlugin):
+                continue
+
             try:
+                self._logger.info("Stopping plugin: %s", plugin.name)
                 plugin.stop()
+                self._logger.info("Successfully stopped plugin: %s", plugin.name)
             except Exception:
                 self._logger.exception("Exception thrown while stopping plugin %s", plugin.name)
 
-    def _is_responsible_for_plugin(self, plugin: BasePlugin) -> bool:
-
-        main_or_isolated_plugin = plugin.process_scope in self.MAIN_AND_ISOLATED_SCOPES
-        shared_plugin = not main_or_isolated_plugin
-
-        manager_for_main_or_isolated = isinstance(self._scope, MainAndIsolatedProcessScope)
-        manager_for_shared = not manager_for_main_or_isolated
-
-        return ((main_or_isolated_plugin and manager_for_main_or_isolated) or
-                (shared_plugin and manager_for_shared))
-
-    def _create_context_for_plugin(self,
-                                   plugin: BasePlugin,
-                                   args: Namespace,
-                                   chain_config: ChainConfig,
-                                   boot_kwargs: Dict[str, Any]) -> PluginContext:
-
-        context: PluginContext = None
-        if plugin.process_scope in self.MAIN_AND_SHARED_SCOPES:
-            # A plugin that runs in a shared process as well as a plugin that overtakes the main
-            # process uses the endpoint of the PluginManager which will either be the main
-            # endpoint or the networking endpoint in the case of Trinity
-            context = PluginContext(self._scope.endpoint)
-        elif plugin.process_scope is PluginProcessScope.ISOLATED:
-            # A plugin that runs in it's own process gets a new endpoint created to get
-            # passed into that new process
-
-            # mypy doesn't know it can only be that scope at this point. The `isinstance`
-            # check avoids adding an ignore
-            if isinstance(self._scope, MainAndIsolatedProcessScope):
-                endpoint = self._scope.event_bus.create_endpoint(plugin.name)
-                context = PluginContext(endpoint)
-        else:
-            Exception("Invariant: unreachable code path")
-
-        context.args = args
-        context.chain_config = chain_config
-        context.boot_kwargs = boot_kwargs
+    async def shutdown(self) -> None:
+        """
+        Asynchronously shut down all started plugins.
+        """
 
-        return context
+        if isinstance(self._scope, MainAndIsolatedProcessScope):
+            raise UnsuitableShutdownError("Use `shutdown_blocking` for instances of this scope")
+
+        self._logger.info("Shutting down PluginManager with scope %s", type(self._scope))
+
+        async_plugins = [
```

Comment:
@carver doing this inline here to not clutter the class with methods that are only used by this method too much. For a moment I thought we could reuse `_stop_plugins()` for both async / sync but since the sync `stop()` needs to be surrounded with a try/catch and is considered a success as soon as `stop()` returns, that isn't really reusable for the async version. I guess its cleaner to handle these separately.